snacks/tests.py

```diff
@@ -1,3 +1,66 @@
+from django.contrib.auth import get_user_model
 from django.test import TestCase
+from django.urls import reverse
+
+from .models import Snack
+
 
 # Create your tests here.
+
+
+class SnackTests(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="tester", email="tester@email.com", password="pass"
+        )
+
+        self.snack = Snack.objects.create(
+            name="pickle",
+            reviewer=self.user,
+        )
+
+    def test_string_representation(self):
+        self.assertEqual(str(self.snack), "pickle")
+
+    def test_snack_content(self):
+        self.assertEqual(f"{self.snack.title}", "pickle")
+        self.assertEqual(f"{self.snack.purchaser}", "tester")
+
+    def test_snack_list_view(self):
+        response = self.client.get(reverse("snack_list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "pickle")
+        self.assertTemplateUsed(response, "snack_list.html")
+
+    def test_snack_detail_view(self):
+        response = self.client.get(reverse("snack_detail", args="1"))
+        no_response = self.client.get("/100000/")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(no_response.status_code, 404)
+        self.assertContains(response, "Purchaser: tester")
+        self.assertTemplateUsed(response, "snack_detail.html")
+
+    def test_snack_create_view(self):
+        response = self.client.post(
+            reverse("snack_create"),
+            {
+                "name": "Chocolate",
+                "purchaser": self.user.id,
```

Comment:
Almost there! Same issue as line 19. The three values must match the model. The "description" key is necessary as well. Because this additional test snack isn't being successfully created, the assertRedirect page isn't the details page as asserted.

Add "Details about Chocolate" to the description so the assert on line 54 passes as well.